src/auth0-session/session/stateful-session.ts

```diff
@@ -0,0 +1,115 @@
+import { CookieSerializeOptions } from 'cookie';
+import createDebug from '../utils/debug';
+import { Config } from '../config';
+import { Cookies } from '../utils/cookies';
+import { AbstractSession, SessionPayload } from './abstract-session';
+import { generateCookieValue, getCookieValue } from '../utils/signed-cookies';
+import { signing } from '../utils/hkdf';
+
+const debug = createDebug('stateful-session');
+
+export interface SessionStore<Session> {
+  /**
+   * Gets the session from the store given a session ID and passes it to `callback`.
+   */
+  get(sid: string): Promise<SessionPayload<Session> | null | undefined>;
+
+  /**
+   * Upsert a session in the store given a session ID and `SessionData`
+   */
+  set(sid: string, session: SessionPayload<Session>): Promise<void>;
+
+  /**
+   * Destroys the session with the given session ID.
+   */
+  delete(sid: string): Promise<void>;
+}
+
+export class StatefulSession<
+  Req,
+  Res,
+  Session extends { [key: string]: any } = { [key: string]: any }
+> extends AbstractSession<Req, Res, Session> {
+  private keys?: Uint8Array[];
+  private store: SessionStore<Session>;
+
+  constructor(protected config: Config, protected Cookies: new () => Cookies) {
+    super(config, Cookies);
+    this.store = config.session.store as SessionStore<Session>;
```

Comment:
I think I had a good reason, let me check...

Reply:
I get errors dong this because `Session` has required fields

<img width="796" alt="image" src="https://user-images.githubusercontent.com/1299658/211590640-e8cc597f-27ac-4473-a223-78822c8af605.png">